nmstate-varlink/nmstate_varlink.py

```diff
@@ -0,0 +1,236 @@
+
+class NmstatePermissionError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstatePermissionError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateConflictError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateConflictError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateLibnmError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateLibnmError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateVerificationError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateVerificationError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateNotImplementedError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateNotImplementedError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateInternalError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateInternalError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateDependencyError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateDependencyError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class ServiceRequestHandler(varlink.RequestHandler):
+    service = service
+
+
+@service.interface("io.nmstate")
+class NmstateService:
+    def Show(self, include_status_data):
+        """
+        Reports the state data on the system
+        """
+        if include_status_data is None:
+            include_status_data = False
+        try:
+            lib_show = libnmstate.show(include_status_data=include_status_data)
+            state = {
+                "state": {
+                    DNS.KEY: lib_show[DNS.KEY],
```

Comment:
Why are you wrapping these values instead of doing something like 

```
state = libnmstate.show(...)
result = {"state": state}
return result
```